index.js

```diff
@@ -0,0 +1,210 @@
+    },
+    "MIT":
+    {
+        "The MIT License": "[![License: MIT](https://img.shields.io/badge/License-MIT-yellow.svg)](https://opensource.org/licenses/MIT)"
+    },
+    "Mozilla":
+    {
+        "Mozilla Public License 2.0": "[![License: MPL 2.0](https://img.shields.io/badge/License-MPL_2.0-brightgreen.svg)](https://opensource.org/licenses/MPL-2.0)"
+    },
+    "Open Data Commons": {
+        "Attribution License (BY)": "[![License: Open Data Commons Attribution](https://img.shields.io/badge/License-ODC_BY-brightgreen.svg)](https://opendatacommons.org/licenses/by/)",
+        "Open Database License (ODbL)": "[![License: ODbL](https://img.shields.io/badge/License-ODbL-brightgreen.svg)](https://opendatacommons.org/licenses/odbl/)",
+    },
+    "Perl":
+    {
+        "The Perl License": "[![License: Artistic-2.0](https://img.shields.io/badge/License-Perl-0298c3.svg)](https://opensource.org/licenses/Artistic-2.0)",
+        "The Artistic License 2.0": "[![License: Artistic-2.0](https://img.shields.io/badge/License-Artistic_2.0-0298c3.svg)](https://opensource.org/licenses/Artistic-2.0)"
+    },
+    "SIL":
+    {
+        "SIL Open Font License 1.1": "[![License: Open Font-1.1](https://img.shields.io/badge/License-OFL_1.1-lightgreen.svg)](https://opensource.org/licenses/OFL-1.1)"
+    },
+    "Unlicense":
+    {
+        "The Unlicense": "[![License: Unlicense](https://img.shields.io/badge/license-Unlicense-blue.svg)](http://unlicense.org/)"
+    },
+    "WTFPL":
+    {
+        "The Do What the Fuck You Want to Public License": "[![License: WTFPL](https://img.shields.io/badge/License-WTFPL-brightgreen.svg)](http://www.wtfpl.net/about/)"
+    },
+    "Zlib":
+    {
+        "The zlib/libpng License": "[![License: Zlib](https://img.shields.io/badge/License-Zlib-lightgrey.svg)](https://opensource.org/licenses/Zlib)"
+    },
+    "None": {
+        "None": ""
+    }
+}
+
+const EMAIL_REGEX = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/;
+
+const ALL_LICENSES_BY_KEYS = {};
+
+const CHOICES = generateChoices();
+
+// function that generates choices array out of the licenses object hard coded above as well as the object from which the badge is fetched
+
+function generateChoices() {
+    var choicesArr = []
+    Object.keys(LICENSES).forEach(type => {
+        choicesArr.push(new inquirer.Separator("----" + type + "----"));
+        var licensesByType = LICENSES[type];
+        Object.assign(ALL_LICENSES_BY_KEYS, licensesByType);
+        Object.keys(licensesByType).forEach(license => {
+            choicesArr.push(license);
+        })
+    })
+    return choicesArr;
+}
+
+// Questions list
+
+const QUESTIONS = [
+    {
+        type: "input",
+        name: "gitUsername",
+        message: "What is your Github Username?"
+    },
+    {
+        type: "input",
+        name: "email",
+        message: "What is your email?",
+        validate: function (input) {
```

Comment:
great addition of validation logic - this is perfect for creating a scalable and robust app!